src/core/ddsc/src/dds_write.c

```diff
@@ -193,27 +193,28 @@ static dds_return_t deliver_locally (struct writer *wr, struct ddsi_serdata *pay
 
 #if DDS_HAS_SHM
 static bool deliver_data_via_iceoryx(dds_writer *wr, struct ddsi_serdata *d) {
-    if (wr->m_iox_pub != NULL && d->iox_chunk != NULL)
-    {
-      iox_chunk_header_t* chunk_header = iox_chunk_header_from_user_payload(d->iox_chunk);
-      iceoryx_header_t *ice_hdr = iox_chunk_header_to_user_header(chunk_header);
-
-      // Local readers go through Iceoryx as well (because the Iceoryx support code doesn't exclude
-      // that), which means we should suppress the internal path
-      ice_hdr->guid = wr->m_wr->e.guid;
-      ice_hdr->tstamp = d->timestamp.v;
-      ice_hdr->statusinfo = d->statusinfo;
-      ice_hdr->data_kind = (unsigned char)d->kind;
-      ddsi_serdata_get_keyhash (d, &ice_hdr->keyhash, false);
-      // iox_pub_publish_chunk takes ownership, storing a null pointer here doesn't
-      // preclude the existence of race conditions on this, but it certainly improves
-      // the chances of detecting them
-
-      iox_pub_publish_chunk (wr->m_iox_pub, d->iox_chunk);
-      d->iox_chunk = NULL;
-      return true; //we published the chunk
-    }
-    return false; // we did not publish the chunk
+  if (wr && wr->m_iox_pub != NULL && d->iox_chunk != NULL) {
```

Comment:
Moved the check for `wr` here, rest is only reformatting (reduce indent whitespace).